### PR TITLE
fix(pie-notification): DSW-1963 address design review notes

### PR DESCRIPTION
**Pull request by raoufswe**
## Describe your changes (can list changeset entries if preferable)

Addresses the design review feedback:
 - The icons size should be 24 x 24px
 - The top padding of the icon is currently 4px, and needs to be reduced to 2px
 - The title should use token 'heading-s/narrow' (16px, 24lh)
 - Update the story title text to `heading` to be consistent
 - The space between the title and supporting text should be 4px
 - The supporting text shouldn't run underneath the close icon.
- When 'stacked actions = true', the leading action should be above.

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] I have added thorough tests where applicable (unit / component / visual)
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] I have reviewed visual test updates properly before approving
- [x] If changes will affect consumers of the package, I have created a changeset entry.
- [x] If a changeset file has been created, I have used the `/snapit` functionality to test my changes in a consuming application

### Reviewer 1 - @xander-marjoram 
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] If there are visual test updates, I have reviewed them

### Reviewer 2
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: packages/components/pie-notification/src/index.ts [hunk 1]
```diff
@@ -64,9 +64,6 @@ export class PieNotification extends LitElement implements NotificationProps {
     @property({ type: Boolean })
     public hideIcon = false;
 
-    @property({ type: Boolean })
-    public hideCloseIcon = false;
```
Comment (raoufswe) [inline packages/components/pie-notification/src/index.ts:68]: prop isn't used.

File: packages/components/pie-notification/src/index.ts [hunk 2]
```diff
@@ -84,19 +81,9 @@ export class PieNotification extends LitElement implements NotificationProps {
     @state()
     protected _hasIconClass = false;
 
-    @state()
-    protected _hasContentGutter = false;
```
Comment (raoufswe) [inline packages/components/pie-notification/src/index.ts:88]: the behavior should be applied for default, thus, removing it and using it directly.

File: packages/components/pie-notification/src/index.ts [hunk 3]
```diff
@@ -84,19 +81,9 @@ export class PieNotification extends LitElement implements NotificationProps {
...
-
     // Renders a `CSSResult` generated from SCSS by Vite
     static styles = unsafeCSS(styles);
 
-    /**
-     * Lifecycle method executed after component renders.
-     */
-    protected firstUpdated (): void {
```
Comment (raoufswe) [inline packages/components/pie-notification/src/index.ts:96]: `firstUpdated` doesn't run on the server, thus, resulting in the following error in nextjs: 

image


we already use `willUpdate` on line 92 to trigger the same behavior and the cycle runs in the server too so it should be enough.

**Review by thejfreitas — APPROVED** [review]
Nice job!

**Review by thejfreitas — APPROVED** [review]

**Review by xander-marjoram — APPROVED** [review]

**Pull request merged**